graphql/schema.js

```diff
@@ -0,0 +1,270 @@
+  name: 'Mutation',
+  description: 'Functions to mutate stuff',
+  fields: {
+    createQuestionnaire: {
+      type: Questionnaire,
+
+      args : {
+        title : {
+          type : new GraphQLNonNull(GraphQLString)
+        },
+        description : {
+          type : new GraphQLNonNull(GraphQLString)
+        },
+        theme : {
+          type : new GraphQLNonNull(GraphQLString)
+        },
+        legalBasis : {
+          type : new GraphQLNonNull(GraphQLString)
+        },
+        navigation : {
+          type : GraphQLBoolean
+        }
+      },
+
+      resolve(source, { title, description, theme, legalBasis, navigation }) {
+        return models.Questionnaire.create({
+          title,
+          description,
+          theme,
+          legalBasis,
+          navigation
+        });
+      }
+    },
+
+    createPage: {
+      type: Page,
+
+      args : {
+        title : {
+          type : new GraphQLNonNull(GraphQLString)
+        },
+        description : {
+          type : new GraphQLNonNull(GraphQLString)
+        },
+        questionnaireId : {
+          type : GraphQLID
+        }
+      },
+
+      resolve(source, { title, description, questionnaireId }) {
+        return models.Page.create({
+          title,
+          description,
+          QuestionnaireId : questionnaireId
+        });
+      }
+    },
+
+    createQuestion: {
+      type: Question,
+
+      args : {
+        title : {
+          type : new GraphQLNonNull(GraphQLString)
+        },
+        description : {
+          type : new GraphQLNonNull(GraphQLString)
+        },
+        guidance : {
+          type : new GraphQLNonNull(GraphQLString)
+        },
+        type : {
+          type : GraphQLString
+        },
+        mandatory : {
+          type : GraphQLBoolean
+        },
+        pageId : {
+          type : GraphQLID
+        }
+      },
+
+      resolve(source, { title, description, guidance, type, mandatory, pageId }) {
+        return models.Question.create({
+          title,
+          description,
+          guidance,
+          type,
+          mandatory,
+          PageId : pageId
+        });
+      }
+    },
+
+    createAnswer: {
+      type: Answer,
+
+      args : {
+        title : {
+          type : new GraphQLNonNull(GraphQLString)
+        },
+
+        description : {
+          type : GraphQLString
+        },
+
+        guidance : {
+          type : GraphQLString
+        },
+
+        label : {
+          type : GraphQLString
+        },
+
+        qCode : {
+          type : GraphQLString
+        },
+
+        type : {
+          type : new GraphQLNonNull(GraphQLString)
+        },
+
+        mandatory : {
+          type : GraphQLBoolean
+        },
+
+        questionId: {
+          type : GraphQLID
+        }
+      },
+
+      resolve(source, { title, description, guidance, label, qCode, type, mandatory, questionId }) {
+        return models.Answer.create({
```

Comment:
For isolating ourselves from sequelize, we're using the graphql-sequelize lib here to generate the schema, so we're always going to bound to it somewhat. We pay for DRY with some coupling.

As for auth, that could/would be handled by an earlier express middleware, so graphql can be completely ignorant of that concern. Resolvers also receive the request object (which would be decorated with user info by that point), so you can then select by `req.user.id` or something like that